lightning/src/util/ser_macros.rs

```diff
@@ -372,7 +372,87 @@ macro_rules! impl_writeable_tlv_based {
 				}, {
 					$(($vectype, $vecfield)),*
 				});
-				_init_tlv_based_struct!({$($reqfield),*}, {$($field),*}, {$($vecfield),*})
+				_init_tlv_based_struct!($st, {$($reqfield),*}, {$($field),*}, {$($vecfield),*})
+			}
+		}
+	}
+}
+
+/// Implement Readable and Writeable for an enum, with struct variants stored as TLVs and tuple
+/// variants stored directly.
+/// The format is, for example
+/// impl_writeable_tlv_based_enum!(EnumName,
+///   (0, VariantName) => { (0, variant_field)}, {(1, variant_optional_field)}, {};
```

Comment:
As an example, yes, I assume future-added optional fields will be odd (ie "if you dont know how to read this, ignore it"), but there's no written rule - the `Option` serialization section means it may or may not be there, but is separate from the "should old versions ignore or consider the object invalid" which is controlled by odd-or-even.